plugins/modules/smbshare.py

```diff
@@ -561,6 +701,47 @@ def validate_umask(self, umask):
             else:
                 umask = int(umask / 10)
 
+    def update_acl_details(self, smb_share_details):
+        acl_params = self.module.params.get("acl")
+        if acl_params:
+            payload = {"add_aces": [], "remove_aces": []}
+            for each in acl_params:
+                data = {"trustee_name": each["trustee_name"],
+                        "trustee_type": each["trustee_type"],
+                        "access_level": each["access_level"],
+                        "access_type": each["access_type"]}
+                if each['state'] == 'present':
+                    payload["add_aces"].append(data)
+                elif each['state'] == 'absent':
+                    payload["remove_aces"].append(data)
+        acl_details = {"aces": []}
+        try:
+            changes, changed = [], False
+            if smb_share_details:
+                acl_details = self.provisioning.get_acl(smb_share_details['smb_share_details']['id'])
+            if acl_params:
+                for each_acl in payload["add_aces"]:
+                    if each_acl not in acl_details["aces"]:
+                        changes.append(True)
+                        break
+                else:
+                    payload["add_aces"] = []
+                for each_acl in payload["remove_aces"]:
+                    if each_acl in acl_details["aces"]:
+                        changes.append(True)
+                        break
+                else:
+                    payload["remove_aces"] = []
+                if any(changes):
+                    self.provisioning.set_acl(smb_share_details['smb_share_details']['id'],
```

Comment:
I guess provisioning file changes are not pushed in the PR

Reply:
I have updated those changes are in SDK(https://github.com/dell/python-powerstore/pull/29/files)